Index: main.py
===================================================================
diff --git a/main.py b/main.py
--- a/main.py	
+++ b/main.py	
@@ -1,55 +1,72 @@
 def main_menu():
     menu()
-    response = input()
-    try:
-        if response == 'add':
-            add()
-        elif response == 'done':
-            done()
-        elif response == 'update':
-            update()
-        elif response == 'list':
-            list_not_done()
-        elif response == 'list-done':
-            list_done()
-        elif response == 'list-all':
-            list_all()
-        elif response == 'quit':
-            return
-        else:
-            raise Exception
+    response = ""
+    tache = "",
+    while response != "quit":
+        response = input()
+        try:
+            if response == 'add':
+                tache = add(tache)
+                print(tache)
+            elif response == 'done':
+                tache = done(tache)
+                print(tache)
+            elif response == 'update':
+                tache = update(tache)
+                print(tache)
+            elif response == 'list':
+                list_not_done(tache)
+            elif response == 'list-done':
+                list_done(tache)
+            elif response == 'list-all':
+                list_all(tache)
+            elif response == 'quit':
+                return
+            else:
+                raise Exception
 
-    except Exception as e:
-        message = "la commande n'existe pas"
-        if e.__class__ == NotImplementedError:
-            message = 'désolé vous ne pouvez pas utiliser cette fonction pour l\'instant'
+        except Exception as e:
+            message = "la commande n'existe pas"
+            if e.__class__ == NotImplementedError:
+                message = 'désolé vous ne pouvez pas utiliser cette fonction pour l\'instant'
 
-        print(message)
-    main_menu()
+            print(message)
+        menu()
 
 
-def add():
-    raise NotImplementedError
+def add(t):
+    t = input("quelle tache souhaitez-vous ajouter ? \n")
+    return t, 0
 
 
-def done():
-    raise NotImplementedError
+def done(t):
+    return t[0], 1
 
 
-def update():
-    raise NotImplementedError
+def update(t):
+    e = input("quel nouveau nom souhaitez-vous ? \n")
+    if t[1] == 1:
+        return e, 1
+    else:
+        return e, 0
 
 
-def list_not_done():
-    raise NotImplementedError
+def list_not_done(t):
+    if t[1] == 0:
+        return print(t[0])
+    else:
+        return print("aucune tache a afficher")
 
 
-def list_done():
-    raise NotImplementedError
+def list_done(t):
+    if t[1] == 1:
+        return print(t[0])
+    else:
+        return print("aucune tache a afficher")
 
 
-def list_all():
-    raise NotImplementedError
+def list_all(t):
+    return print(t[0])
 
 
 def menu():
